### PR TITLE
Creates PPM move document summary handler [delivers #159120642]

### PR DESCRIPTION
**Pull request by stangah**
## Description

Given a PPM ID, this handler will assemble the orders document and an assortment of linked `MoveDocuments` and convert/merge into a single PDF.

Assuming you have some move documents uploaded (and a real orders document, e2e data load doesn't work), you can test this via SwaggerUI: internal/docs#/ppm/createPPMAttachments

Still needs:
[ ] tests

## Reviewer Notes

I'd love more thought about which HTTP status codes should be returned when. It's common at this point for malformed PDF files to blow up the whole operation. I have this returning 422 unprocessable entity, but maybe there's something better?

## Code Review Verification Steps

* [ ] End to end tests pass (`make e2e_test`).
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/159120642) for this change

### CODE REVIEW

File: pkg/handlers/handlers.go [hunk 1]
```diff
@@ -147,6 +147,7 @@ func NewInternalAPIHandler(context HandlerContext) http.Handler {
 	internalAPI.PpmShowPPMSitEstimateHandler = ShowPPMSitEstimateHandler(context)
 	internalAPI.PpmShowPPMIncentiveHandler = ShowPPMIncentiveHandler(context)
 	internalAPI.PpmRequestPPMPaymentHandler = RequestPPMPaymentHandler(context)
+	internalAPI.PpmCreatePPMSummaryHandler = CreatePersonallyProcuredMoveSummaryHandler(context)
```
Comment (tinyels) [inline pkg/handlers/handlers.go]: I find the name PPM Summary a little misleading. Could we call it PPM Receipts or Attachments instead?

File: pkg/paperwork/generator.go [hunk 2]
```diff
@@ -29,24 +28,35 @@ const (
...
 // NewGenerator creates a new Generator.
 func NewGenerator(db *pop.Connection, logger *zap.Logger, uploader *uploader.Uploader) (*Generator, error) {
-	directory, err := ioutil.TempDir("", "generator")
+	var fs = afero.NewMemMapFs()
+	afs := &afero.Afero{Fs: fs}
+
+	pdfConfig := pdfcpu.NewInMemoryConfiguration()
+	pdfConfig.FileSystem = fs
+
+	// directory, err := ioutil.TempDir("", "generator")
```
Comment (jim) [inline pkg/paperwork/generator.go]: This comment can be deleted.

File: pkg/paperwork/generator.go [hunk 3]
```diff
@@ -55,61 +65,99 @@ type inputFile struct {
 	ContentType string
 }
 
-func (g *Generator) newTempFile() (*os.File, error) {
-	outputFile, err := ioutil.TempFile(g.workDir, "temp")
+func (g *Generator) newTempFile() (afero.File, error) {
+	// outputFile, err := ioutil.TempFile(g.workDir, "temp")
```
Comment (jim) [inline pkg/paperwork/generator.go]: Likewise 😄

File: pkg/paperwork/generator.go [hunk 4]
```diff
@@ -29,24 +28,35 @@ const (
...
 }
 
 // NewGenerator creates a new Generator.
 func NewGenerator(db *pop.Connection, logger *zap.Logger, uploader *uploader.Uploader) (*Generator, error) {
-	directory, err := ioutil.TempDir("", "generator")
+	var fs = afero.NewMemMapFs()
+	afs := &afero.Afero{Fs: fs}
+
+	pdfConfig := pdfcpu.NewInMemoryConfiguration()
+	pdfConfig.FileSystem = fs
```
Comment (pjdufour-truss) [inline pkg/paperwork/generator.go]: I think you could change to just this:

```
pdfConfig := pdfcpu.NewInMemoryConfiguration()
afs := &afero.Afero{Fs: pdfConfig.FileSystem}
```

Comment (stangah) [reply pkg/paperwork/generator.go]: You're right, but I kind of like the idea of keeping control over setting the fs being used, even though it's your fork and we both know exactly what it is :p

**Pull request merged**